library/spdm_common_lib/libspdm_com_opaque_data.c

```diff
@@ -192,21 +191,24 @@ libspdm_build_opaque_data_supported_version_data(libspdm_context_t *spdm_context
     return LIBSPDM_STATUS_SUCCESS;
 }
 
+
+
 /**
  * Process opaque data supported version.
  *
- * This function should be called in KEY_EXCHANGE/PSK_EXCHANGE request parsing in responder.
+ * This function should be called in
+ * libspdm_process_opaque_data_supported_version_data/libspdm_process_opaque_data_version_selection_data.
  *
- * @param  data_in_size                   size in bytes of the data_in.
- * @param  data_in                       A pointer to the buffer to store the opaque data supported version.
+ * @param  data_in_size                    size in bytes of the data_in.
+ * @param  data_in                         A pointer to the buffer to store the opaque data
+ * @param[in]  process_supported_version   When true, process opaque data supported version;
+ *                                         When false, process opaque data selection version.
  *
  * @retval RETURN_SUCCESS               The opaque data supported version is processed successfully.
  * @retval RETURN_UNSUPPORTED           The data_in is NOT opaque data supported version.
  **/
-libspdm_return_t
-libspdm_process_opaque_data_supported_version_data(libspdm_context_t *spdm_context,
-                                                   size_t data_in_size,
-                                                   const void *data_in)
+libspdm_return_t libspdm_process_opaque_data(libspdm_context_t *spdm_context, size_t data_in_size,
+                                             const void *data_in, bool process_supported_version)
```

Comment:
This is not a good idea. What if there are more opaque data need to handle? 
A BOOL cannot handle more than 2 state.

Why not take my feedback to create a common function to element based upon ID ?